File: core/src/main/java/xtdb/vector/IValueWriter.java
package xtdb.vector;

import clojure.lang.Keyword;

import java.nio.ByteBuffer;

public interface IValueWriter {

    void writeNull();

    void writeBoolean(boolean booleanValue);

    void writeByte(byte byteValue);

    void writeShort(short shortValue);

    void writeInt(int intValue);

    void writeLong(long longValue);

    void writeFloat(float floatValue);

    void writeDouble(double doubleValue);

    void writeBytes(ByteBuffer bytesValue);

    void writeObject(Object objectValue);

    IValueWriter legWriter(Keyword leg);
}

File: core/src/main/kotlin/xtdb/vector/IValueWriter.kt
package xtdb.vector

import clojure.lang.Keyword
import java.nio.ByteBuffer

interface IValueWriter {
    fun writeNull()
    fun writeBoolean(v: Boolean)
    fun writeByte(v: Byte)
    fun writeShort(v: Short)
    fun writeInt(v: Int)
    fun writeLong(v: Long)
    fun writeFloat(v: Float)
    fun writeDouble(v: Double)
    fun writeBytes(v: ByteBuffer)
    fun writeObject(v: Any?)

    fun legWriter(leg: Keyword): IValueWriter
}
